test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua

```diff
@@ -0,0 +1,309 @@
+-- 1. User choose <App> in the list of registered aps on HMI.
+-- 2. The User allows definite permissions.
+--
+-- Expected:
+-- 1. HMI->SDL: SDL.ActivateApp {appID}
+-- SDL->HMI: SDL.ActivateApp_response{isPermissionsConsentNeeded: true, params}
+-- HMI->SDL: GetUserFriendlyMessage{params},
+-- SDL->HMI: GetUserFriendlyMessage_response{params}
+-- HMI->SDL: GetListOfPermissions{appID}
+-- SDL->HMI: GetListOfPermissions_response{}
+-- 2. HMI->SDL: OnAppPermissionConsent {allowed = false}
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+--ToDo: shall be removed when issue: "ATF does not stop HB timers by closing session and connection" is fixed
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require ('user_modules/shared_testcases/commonSteps')
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+local mobile_session = require('mobile_session')
+
+--[[ Local Functions ]]
+local function BackupPreloaded()
+  os.execute('cp ' .. config.pathToSDL .. 'sdl_preloaded_pt.json' .. ' ' .. config.pathToSDL .. 'backup_sdl_preloaded_pt.json')
+  os.execute('rm ' .. config.pathToSDL .. 'policy.sqlite')
+end
+
+local function RestorePreloadedPT()
+  os.execute('rm ' .. config.pathToSDL .. 'sdl_preloaded_pt.json')
+  os.execute('cp ' .. config.pathToSDL .. 'backup_sdl_preloaded_pt.json' .. ' ' .. config.pathToSDL .. 'sdl_preloaded_pt.json')
+end
+
+local function SetPermissionsForPre_DataConsent()
+  local pathToFile = config.pathToSDL .. 'sdl_preloaded_pt.json'
+  local file = io.open(pathToFile, "r")
+  local json_data = file:read("*all") -- may be abbreviated to "*a";
+  file:close()
+  local json = require("modules/json")
+  local data = json.decode(json_data)
+
+  if data.policy_table.functional_groupings["DataConsent-2"] then
+    data.policy_table.functional_groupings["DataConsent-2"] = {rpcs = null}
+  end
+  -- set for group in pre_DataConsent section permissions with RPCs and HMI levels for them
+  data.policy_table.functional_groupings[data.policy_table.app_policies.default.groups[1]] = {rpcs = {
+      OnHMIStatus =
+      {hmi_levels = {"BACKGROUND", "FULL", "LIMITED", "NONE"}},
+      OnPermissionsChange =
+      {hmi_levels = {"BACKGROUND", "FULL", "LIMITED", "NONE"}},
+      OnSystemRequest =
+      {hmi_levels = {"BACKGROUND", "FULL", "LIMITED", "NONE"}},
+      SystemRequest =
+      {hmi_levels = {"BACKGROUND", "FULL", "LIMITED", "NONE"}},
+  }}
+  data = json.encode(data)
+  file = io.open(pathToFile, "w")
+  file:write(data)
+  file:close()
+end
+
+--[[ Precondition ]]
+local function Precondition()
+
+  commonFunctions:userPrint(34, "-- Precondition --")
+
+  function Test:StopSDL()
```

Comment:
Please add to names of all functions here prefix Precondition_
in this case 
function Test:Precondition_StopSDL()